content/blog/2020/2020-10-01-document-embedders.md

```diff
@@ -0,0 +1,72 @@
++++
+author = "Primož Godec and Nikola Đukić"
+date = "2020-10-01"
+draft = false
+title = "How to identify fake news with document embeddings"
+type = "blog"
+thumbImage = "/blog_img/2020/2020-10-01-document-embedding-widget.png"
+frontPageImage = "/blog_img/2020/2020-10-01-document-embedding-widget.png"
+categories = ["text mining", "corpus", "classification"]
+shortExcerpt = "New Document embedder widget and its use for classification"
+longExcerpt = "Presenting document embeddings widget and how to identify fake news."
+x2images = true  # true if using retina screenshots, else false
++++
+
+Text is by default described by the sequence of character. Since every machine learning algorithm needs numbers, we need to transform it into vectors of real numbers before we can continue with the analysis. To do this, we can use various approaches. Orange currently offers bag-of-words approach and now also Document embedding by fastText. 
+In this post, we explain what document embedding is, why it is useful, and show its usage on the classification example.
+
+## Word embedding and document embedding
+
+Before we understand document embeddings, we need to understand the concept of word embeddings. Word embedding is a representation of a word in multidimensional spaces such that words with similar meanings have similar embedding. It means that each word is mapped to the vector of real numbers that represents the word.
+Embedding models are mostly based on neural networks.
+
+Document embedding is computed in two steps. First, each word is embedded with the word embedding then word embeddings are aggregated. The most common type of aggregation is the average over each dimension.
+
+## Why and when should we use embedders?
+
+Compared to [bag-of-words](/widget-catalog/text-mining/bagofwords-widget), which counts the number of appearances of each token in the document, embeddings have two main advantages:
+
+- **They do not have a dimensionality problem** The result of bag-of-words is a table which has the number of features equal to the number of unique tokens in all documents in a corpus. Large corpora with long texts result in a large number of unique tokens. It results in huge tables which can exceed memory in the computer. Huge tables also increase the learning and evaluation time of machine learning models. Embedders have constant dimensionality of the vector, which is 300 for fastText embeddings that Orange uses. 
+- **Most of the preprocessing is not required** In case of bag-of-words approach, we solve the dimensionality problem with the text preprocessing where we remove tokens (e.g. words) that seem to be less important for the analysis. It can also cause the removal of some important tokens. When using embedders, we do not need to remove tokens, so we are not losing the accuracy this way. Also most of the basic preprocessing can be omitted (such as normalization) in case of fastText embeddings.
+
```

Comment:
I'd also add something like:
**They can be pretrained** 
Word embedding models can be pretrained on large corpora with billions of tokens. That way they capture the significant characteristics of the language and produce the embeddings of high quality. Pretrained models are then used to obtain embeddings of smaller datasets. Our Document Embedding widget uses pretrained fastText models and is suitable for corpora of any size.

Reply:
Added, thank you